Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -493,17 +493,17 @@ inline bool ShouldDraw() {
         return SUCCEEDED(_firstErrorHr) && CurrentGState().ShouldDraw();
     }
 
-    inline void PushBeginDraw() {
-        if ((_beginEndDrawDepth)++ == 0) {
+    inline void PushBeginDraw(bool forceBeginDraw = false) {
+        if ((_beginEndDrawDepth)++ == 0 || forceBeginDraw) {
```

Comment:
Given forceEndDraw can be used by anyone, i would prefer the counter to be manipulated.
regardless of manipulation, if someone does not start a PushBeginDraw.forceEndDraw after doing a PopEndDraw.forceEndDraw, we'll get into an invalid state.
So again it depends on how it's being used.

---
In reply to: [105750699](https://github.com/Microsoft/WinObjC/pull/2208#discussion_r105750699) [](ancestors = 105750699)